pkg/workflows/management/upgrade_curated_packages.go

```diff
@@ -0,0 +1,69 @@
+package management
+
+import (
+	"context"
+
+	appsv1 "k8s.io/api/apps/v1"
+
+	"github.com/aws/eks-anywhere/pkg/clients/kubernetes"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/task"
+	"github.com/aws/eks-anywhere/pkg/utils/ptr"
+	"github.com/aws/eks-anywhere/pkg/workflows"
+)
+
+type upgradeCuratedPackagesTask struct{}
+
+func (s *upgradeCuratedPackagesTask) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+	if commandContext.CurrentClusterSpec.Cluster.Spec.RegistryMirrorConfiguration.Equal(commandContext.ClusterSpec.Cluster.Spec.RegistryMirrorConfiguration) {
+		return nil
+	}
+
+	commandContext.PackageInstaller.InstallCuratedPackages(ctx)
+
+	client, err := commandContext.ClientFactory.BuildClientFromKubeconfig(commandContext.ManagementCluster.KubeconfigFile)
+	if err != nil {
+		commandContext.SetError(err)
+	}
+
+	packagesManager := &appsv1.Deployment{}
+
+	err = client.Get(ctx, "eks-anywhere-packages", constants.EksaPackagesName, packagesManager)
+	if err != nil {
+		commandContext.SetError(err)
+		return &workflows.CollectMgmtClusterDiagnosticsTask{}
+	}
+
+	replicas := packagesManager.Spec.Replicas
+	packagesManager.Spec.Replicas = ptr.Int32(0)
+	err = client.ApplyServerSide(ctx, constants.DefaultCLIFieldManager, packagesManager, kubernetes.ApplyServerSideOptions{ForceOwnership: true})
```

Comment:
Can't we use a patch instead of server side? Server side apply can still overwrite changes if the resource is written in between our read and apply

Patch guarantees we only update the field we want to update (replicas)